### ISSUE TITLE
context.SubFields does not include fragment fields on a NonNull field

### ISSUE THREAD

**Issue by pvenable** [graphql-dotnet/graphql-dotnet#1123]
## Description

`context.SubFields` is not being populated with fragment fields on NonNullGraphType fields.  It works fine for nullable fields.

### Steps to reproduce

Here's a failing test exposing the issue (drops right into `StarWarsSubFieldsTest.cs`):

```
        [Fact]
        public void subfields_contains_keys_from_fragments_on_non_null_fields()
        {
            RootQuery.Field<NonNullGraphType<HumanType>>("luke", resolve: context =>
            {
                context.SubFields.ShouldNotBeNull();
                context.SubFields.Keys.ShouldContain("id");
                context.SubFields.Keys.ShouldContain("name");
                return new Human { Id = "1", Name = "Luke" };
            });

            var query = @"
                query Luke {
                    luke {
                        ...HumanData
                    }
                }

                fragment HumanData on Human {
                    id
                    name
                }
            ";

            var expected = @"
                {
                    luke: {
                        id: ""1"",
                        name: ""Luke""
                    }
                }
            ";

            AssertQuerySuccess(query, expected);
        }
```

(Note: The same test with `Field<HumanType>` instead of `Field<NonNullGraphType<HumanType>>` will already pass.)

### Expected result

The fragment field keys should be contained in `context.SubFields.Keys`.

### Actual result

The fragment field keys are not included in `context.SubFields.Keys`.

**Comment by pvenable** [comment]
A possible fix would be adding something like this to `ExecutionHelper.DoesFragmentConditionMatch`:

```
            if (type is NonNullGraphType nonNullGraphType)
            {
                return conditionalType.Equals(nonNullGraphType.ResolvedType);
            }
```

I'm not sure if that's the right/best place to check for `NonNullGraphType`, but I believe this would resolve the issue.

**Comment by sungam3r** [comment]
@pvenable Thanks for the repro. Fixed in #2308. The same error was in the case of using `ListGraphType`.

**Comment by sungam3r** [comment]
Will be published in v4.
